code/datums/ai/basic_mobs/basic_ai_behaviors/basic_attacking.dm

```diff
@@ -82,3 +88,52 @@
 	. = ..()
 	if(!succeeded)
 		controller.clear_blackboard_key(target_key)
+
+/datum/ai_behavior/basic_ranged_attack/proc/check_friendly_in_path(mob/living/source, atom/target, datum/targetting_datum/targetting_datum)
+	var/list/turfs_list = calculate_trajectory(source, target)
+	for(var/turf/possible_turf in turfs_list)
```

Comment:
is `turfs_list` guaranteed to only have subtypes of `/turf`? if so, add `as anything`. if calculate_trajectory doesn't just return a list of `/turf`s, disregard